File: frontend/src/app/core/services/categoria.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CategoriaRequest, CategoriaResponse } from '../models/categoria.model';

@Injectable({
    providedIn: 'root'
})
export class CategoriaService {
    private readonly apiUrl = 'http://localhost:8080/api/categorias';

    constructor(private http: HttpClient) {}

    getAll(): Observable<CategoriaResponse[]> {
        return this.http.get<CategoriaResponse[]>(this.apiUrl);
    }

    create(payload: CategoriaRequest): Observable<CategoriaResponse> {
        return this.http.post<CategoriaResponse>(this.apiUrl, payload);
    }

    update(id: number, payload: CategoriaRequest): Observable<CategoriaResponse> {
        return this.http.put<CategoriaResponse>(`${this.apiUrl}/${id}`, payload);
    }

    delete(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}`);
    }
}

File: src/main/java/com/josequintero/taskflow/service/CategoriaService.java
package com.josequintero.taskflow.service;

import com.josequintero.taskflow.dto.categoria.CategoriaRequestDto;
import com.josequintero.taskflow.dto.categoria.CategoriaResponseDto;

import java.util.List;

public interface CategoriaService {

    List<CategoriaResponseDto> getAll();

    CategoriaResponseDto create(CategoriaRequestDto request);

    CategoriaResponseDto update(Long id, CategoriaRequestDto request);

    void delete(Long id);
}
